client_end_user/shared/src/commonMain/kotlin/presentation/resturantDetails/RestaurantScreen.kt

```diff
@@ -0,0 +1,257 @@
+package presentation.resturantDetails
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Divider
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import cafe.adriel.voyager.navigator.Navigator
+import com.beepbeep.designSystem.ui.theme.Theme
+import domain.entity.PriceLevel
+import org.jetbrains.compose.resources.ExperimentalResourceApi
+import org.jetbrains.compose.resources.painterResource
+import presentation.auth.login.LoginScreen
+import presentation.base.BaseScreen
+import presentation.composable.BottomSheet
+import presentation.composable.BpImageCard
+import presentation.composable.BpPriceLevel
+import presentation.composable.RatingBar
+import presentation.composable.SectionHeader
+import presentation.composable.modifier.noRippleEffect
+import presentation.resturantDetails.Composable.Chip
+import presentation.resturantDetails.Composable.NeedToLoginSheet
+import resources.Resources
+
+object RestaurantScreen :
+    BaseScreen<RestaurantScreenModel, RestaurantUIState, RestaurantUIEffect, RestaurantInteractionListener>() {
+
+
+    override fun onEffect(effect: RestaurantUIEffect, navigator: Navigator) {
+        when (effect) {
+            is RestaurantUIEffect.onBack -> navigator.pop()
+            is RestaurantUIEffect.onGoToDetails -> {}
+            is RestaurantUIEffect.onGoToLogin -> navigator.push(LoginScreen())
+        }
+    }
+
+    @Composable
+    override fun Content() {
+        initScreen(getScreenModel())
+    }
+
+    @OptIn(ExperimentalResourceApi::class, ExperimentalMaterial3Api::class)
+    @Composable
+    override fun onRender(state: RestaurantUIState, listener: RestaurantInteractionListener) {
+            Box(
```

Comment:
245 line of code is not clean in one screen , please could you seperate components in different composoables ?